service/controller/resource/clusterconfigmap/desired.go

```diff
@@ -215,6 +215,12 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) ([]*cor
 			"domain": r.registryDomain,
 		},
 	}
+	// disable kubernetes client cache for EKS cluster
+	if key.IsEKS(cr) {
```

Comment:
`IsEKS()`  check the ClusterCR for the ControlPlane Reference Kind and Infrastructure Reference Kind so should work also for imported EKS clusters